include/boost/beast/core/impl/basic_stream.hpp

```diff
@@ -161,12 +160,21 @@ close()
 //------------------------------------------------------------------------------
 
 template<class Protocol, class Executor, class RatePolicy>
+template<class Executor2>
```

Comment:
I'm not generally opposed to this change, but aren't we just trading one instantiation for another?

Reply:
`bind_executor` has to instantiate at least 3 additional classes (due to support for binding executors for CompletionTokens) and adds a few hundred lines of header material.